RecoParticleFlow/PFClusterProducer/plugins/SpikeAndDoubleSpikeCleaner.cc

```diff
@@ -117,10 +117,65 @@ clean(const edm::Handle<reco::PFRecHitCollection>& input,
     }    
     const spike_cleaning& clean = _thresholds.find(hitlayer)->second;    
     if( rechit.energy() < clean._singleSpikeThresh ) continue;
+
+
+
+    //Fix needed for HF.  Here, we find the (up to) five companion rechits 
+    //to work in conjunction with the neighbours4() implementation below for the full HF surrounding energy
+    float compsumE = 0.0;
+    if ((hitlayer==PFLayer::HF_EM || hitlayer==PFLayer::HF_HAD))
+	{
+
+	int comp = 1;
+    	if (hitlayer==PFLayer::HF_EM) comp = 2;
+
+	int predphi =2;
+
+	//int comp1 = std::abs(hitlayer-13);
+
+	const HcalDetId& detid = (HcalDetId)rechit.detId();
+	std::vector<uint32_t> rawDetIds;
+	int heta = detid.ieta();
+	int hphi = detid.iphi();
+	
+	//At eta>39, phi segmentation changes 
+	if (std::abs(heta)>39) predphi=4;
+
+ 	int curphiL = hphi-predphi;
+	int curphiH = hphi+predphi;
+
+	//HcalDetId valid phi range (1-72)
+	if (predphi>abs(hphi) or predphi>abs(72-hphi))
```

Comment:
Unnecessary multiple checks (which are even hard to read...)
Why not simply add the two while loop right after L144 and L145, still keeping the comment that explains what they are doing?
(By the way: why using the macro "or", instead of "//" as everywhere else in this code?)